packages/valory/skills/price_estimation_abci/rounds.py

```diff
@@ -144,6 +154,61 @@ class TxHashRound(CollectSameUntilThresholdRound):
     selection_key = get_name(SynchronizedData.most_voted_tx_hash)
 
 
+class DataHashSignRound(CollectDifferentUntilAllRound):
+    """A round in which agents sign the data hash"""
+
+    payload_class = SignaturePayload
+    payload_attribute = "signature"
+    synchronized_data_class = SynchronizedData
+    done_event = Event.DONE
+    none_event = Event.NONE
+    no_majority_event = Event.NO_MAJORITY
+    collection_key = get_name(SynchronizedData.service_data_signatures)
+
+    def check_payload(self, payload: BaseTxPayload) -> None:
+        """Check Payload"""
+        if payload.round_count != self.synchronized_data.round_count:
+            raise TransactionNotValidError(
+                f"Expected round count {self.synchronized_data.round_count} and got {payload.round_count}."
+            )
+
+        if payload.sender in self.collection:
+            raise TransactionNotValidError(
+                f"sender {payload.sender} has already sent value for round: {self.round_id}"
+            )
+
+    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
+        """Process the end of the block."""
+        if self.collection_threshold_reached:
+            signatures = {
+                addr: payload.signature
+                for addr, payload in cast(
+                    Dict[str, SignaturePayload], self.collection
+                ).items()
+            }
+            synchronized_data = self.synchronized_data.update(
+                **{self.collection_key: signatures},
+                synchronized_data_class=SynchronizedData,
+            )
+            return synchronized_data, Event.DONE
+        return None
+
+
+class DataHashSignaturesStoreRound(CollectionRound):
+    """A round in which agents store signatures collected in the DataHashSignRound."""
+
+    payload_class = EmptyPayload
+    payload_attribute = "signature"
+    synchronized_data_class = SynchronizedData
+
+    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
+        """Process the end of the block."""
+        return (
+            self.synchronized_data,
+            Event.DONE,
+        )  # pragma: nocover  # nothing to check here
```

Comment:
So since we are not checking for a payload here, no need to send it in the behaviour.

Reply:
good point

Reply:
stay as is, `return None` does not work here

Reply:
So i was just saying that you dont need to send the payload in the behaviour, the return was good here.